loopchain/blockchain/blocks/v1_0/block_verifier.py

```diff
@@ -1,25 +1,137 @@
+            prev_next_reps_hash = prev_block.header.revealed_next_reps_hash
+            if prev_next_reps_hash:
+                if prev_next_reps_hash != header.validators_hash:
+                    raise RuntimeError(
+                        f"Block({header.height}, {header.hash.hex()}, "
+                        f"RepsHash({header.validators_hash}), "
+                        f"Expected({prev_next_reps_hash}).")
+                prev_reps = self._reps_getter(prev_block.header.validators_hash)
+            else:
+                prev_reps = self._reps_getter(block.header.validators_hash)
+            self.verify_prev_votes(block, prev_reps)
+
+    def verify_prev_votes(self, block: 'Block', prev_reps: Sequence['ExternalAddress']):
+        prev_votes: 'BlockBody' = block.body.prev_votes
+
+        if prev_votes:
+            for vote in prev_votes:
+                if vote and vote.version <= '0.5':
+                    return self.__verify_prev_votes_by_siever(block, prev_reps, vote)
+                self._verify_vote(vote)
+
+    def __verify_prev_votes_by_siever(self, block: 'Block', prev_reps: Sequence['ExternalAddress'], any_vote):
+        header: 'BlockHeader' = block.header
+        body: 'BlockBody' = block.body
+        round_ = 0
+        votes_class = None
+        if any_vote:
+            votes_class = v0_5.BlockVotes if any_vote.version else v0_1a.BlockVotes
+            round_ = any_vote.round
+
+        prev_votes = votes_class(
+            prev_reps, conf.VOTING_RATIO, header.height - 1, round_, header.prev_hash, body.prev_votes)
+        if prev_votes.get_result() is not True:
+            raise RuntimeError(f"Block({header.height}, {header.hash.hex()}, PrevVotes {body.prev_votes}")
+        try:
+            prev_votes.verify()
+        except Exception as e:
+            # FIXME : votes.verify does not verify all votes when raising an exception.
+            raise e
+        return True
+
+    def _verify_vote(self, vote: 'BlockVote'):
+        invoke_result = self._invoke_pool.get_invoke_data(vote.epoch_num, vote.round_num)
+
+        if invoke_result.state_hash != vote.state_hash:
+            raise RuntimeError(
+                f"Block({vote.height}, {vote.data_id.hex()}, "
+                f"StateRootHash({vote.state_hash.hex()}), "
+                f"Expected({invoke_result.state_hash.hex()}).")
+
+        if invoke_result.receipt_hash != vote.receipt_hash:
+            raise RuntimeError(
+                f"Block({vote.height}, {vote.data_id.hex()}, "
+                f"ReceiptRootHash({vote.receipts_hash.hex()}), "
+                f"Expected({invoke_result.receipts_hash.hex()}.")
+
+        if invoke_result.next_validators_hash != vote.next_validators_hash:
+            raise RuntimeError(
+                f"Block({vote.height}, {vote.data_id.hex()}, "
+                f"NextValidatorsHash({vote.next_validators_hash.hex()}), "
+                f"Expected({invoke_result.next_validators_hash.hex()}.")
```

Comment:
LFT expects BlockVote to implement consensus_id.
And I created it by using 4 hashes which are consisted of hashes above.
And LFT passes vote verification only after consensus_id is verified (matched)

So I think this is redundant step.